pkg/controller/delay.go

```diff
@@ -31,7 +31,7 @@ func cancelDelayedDownscaleIfConfigured(ctx context.Context, logger log.Logger,
 		return
 	}
 
-	endpoints := createPrepareDownscaleEndpoints(sts.Namespace, sts.GetName(), 0, int(replicas), prepareURL)
+	endpoints := createPrepareDownscaleEndpoints(sts.Namespace, sts.Spec.ServiceName, 0, int(replicas), prepareURL)
```

Comment:
Both the StatefulSet name and the service name are required to construct the FQDN. It doesn't work to swap one for the other since they can be different.